pkg/cloudinstances/cloud_instance_group.go

```diff
@@ -87,7 +87,7 @@ func GetNodeMap(nodes []v1.Node) map[string]*v1.Node {
 	nodeMap := make(map[string]*v1.Node)
 	for i := range nodes {
 		node := &nodes[i]
-		nodeMap[node.Spec.ExternalID] = node
+		nodeMap[node.Spec.ProviderID] = node
```

Comment:
This one isn't obviously safe to me - do you want to pull this one into a separate PR maybe?